tests/testsuite/run.rs

```diff
@@ -386,6 +387,94 @@ fn run_example() {
     );
 }
 
+#[test]
+fn run_example_autodiscover() {
+    let p = project("foo")
+        .file(
+            "examples/a.rs",
+            r#"
+            fn main() { println!("example"); }
+        "#,
+        )
+        .file(
+            "examples/do_magic.rs",
+            r#"
+            fn main() { println!("magic example"); }
+        "#,
+        );
+
+    if cargotest::is_nightly() {
```

Comment:
This I believe can be avoided with `masquerade_as_nightly_cargo()` for Cargo, and there should be one `#[test]` function per test (so split this function into two).

This test may still need to be at the top of the function though because `--edition` doesn't exist on stable rustc, but it'll just be a prelude to all these tests.

Reply:
will do

Reply:
actually can you explain again? the only part I'm clear on is splitting the test.

what I'm testing here is:in rust nightly + cargo edition feature + edition 2018